### PR TITLE
The Crashed Starwalker

### PR DESCRIPTION
**Pull request by spockye**
## About The Pull Request
This PR adds a beach ruin based around a ship I've previously made, called the "Starwalker"

it contains:
some medical supplies ( oinment slurry / herbal pack / crew monitor / health scanner / charcoal bottle / misc pills )
one Swat suit
one shotgun / one energy cutlass
goliath cloak  / military rig
3 abandoned crates
1 gold crate / one silver crate
lizard wine
one baby carp
a radiant dance machine
a sci protolathe
misc salvage

Lore bit:
After a "most excellent robbery that went like, totally as planned", our protagonists aboard the Starwalker fled the crime scene, with heavy damage to the ship's hull. With one of the Engine blocks almost falling off, The valiant crew decided that the best course of action would be a "Totally rad emergency landing". This, of course, ended in disaster, as the pilot was high on LSD. 
The pilot did however manage to steer them towards a nearby lak- sike, it's just some shallow water. Crashing directly onto the ground, the ship split into multiple fragments, Killing the pilot and crewmate, and Impaling the captain. 
The captain knew that he didn't have long until the bloodloss would get to him, and started moving all his treasure into a nearby cavern. _THERE'S NO WAY_ he would die in that godforsaken ship, nor without his treasures. This is where you now find him, rotting in his "100% real Cow skin" throne _(spacemart Brand Comfy chair)_ . 

- [x] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game
currently there's a bit of a lack in beach ruins, something that I'd like to help resolve!

## Changelog

:cl:
add: Adds a new Beach ruin, the beach_crashed_starwalker
/:cl:

### CODE REVIEW

**Comment by spockye** [comment]
uhh, something's weird

**Comment by spockye** [comment]
I have left your realm, mortals. my PR's exist beyond the limits of your barbaric machines. Look upon my PR's and despair, for you shall never witness their beauty  //edit : Fuck

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm [hunk 1]
```diff
@@ -0,0 +1,5515 @@
...
+	dir = 9
+	},
+/obj/machinery/atmospherics/pipe/simple/supply/hidden/layer2{
+	dir = 10
+	},
+/obj/structure/cable{
+	icon_state = "4-8"
+	},
+/obj/item/gps{
+	gpstag = "NTREC1";
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm]: iirc this is the kilo's tag too. I'd give em a different one

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm [hunk 2]
```diff
@@ -0,0 +1,5515 @@
...
+/area/ruin/unpowered)
+"it" = (
+/obj/effect/turf_decal/industrial/warning{
+	dir = 8
+	},
+/obj/structure/closet/body_bag,
+/obj/effect/decal/cleanable/dirt,
+/obj/effect/decal/cleanable/dirt,
+/obj/item/circuitboard/computer/rdconsole,
+/obj/item/circuitboard/machine/protolathe/department/science{
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm]: it's just a circuitboard but is the science one appropriate for this ruin? I'd say like... Peacekeeping for ammo (pirate ship) or a cargo one.

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm [hunk 3]
```diff
@@ -0,0 +1,5515 @@
...
+	},
+/obj/structure/grille,
+/turf/open/floor/engine/hull,
+/area/ruin/beach/starwalker)
+"nw" = (
+/mob/living/simple_animal/hostile/carp{
+	resize = 0.5;
+	melee_damage_lower = 0;
+	melee_damage_upper = 0;
+	name = "Baby carp"
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm]: give this guy a desc too

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm [hunk 4]
```diff
@@ -0,0 +1,5515 @@
...
+	dir = 4
+	},
+/obj/machinery/power/apc/auto_name/west,
+/obj/structure/cable/cyan{
+	icon_state = "0-4"
+	},
+/turf/open/floor/plasteel/stairs{
+	barefootstep = "woodbarefoot";
+	color = "#A47449";
+	footstep = "wood"
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm]: you should subtype the "wood" stairs, maybe get a sprite

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm [hunk 5]
```diff
@@ -0,0 +1,5515 @@
...
+/obj/structure/cable/green{
+	icon_state = "1-6"
+	},
+/obj/structure/spacevine,
+/turf/open/floor/plasteel/patterned,
+/area/ruin/beach/starwalker)
+"Xa" = (
+/obj/effect/decal/cleanable/dirt/dust,
+/obj/effect/decal/cleanable/dirt,
+/obj/effect/decal/cleanable/dirt,
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm:3275]: doubled up dirt

**Review by Erikafox — COMMENTED** [review]
map review later

File: _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm (hunk 5 above)
Comment (spockye) [reply _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm:3275]: does that cause issues?

Comment (Erikafox) [reply _maps/RandomRuins/BeachRuins/beach_crashed_starwalker.dmm:3275]: No

**Review by Erikafox — CHANGES_REQUESTED** [review]
image
scatter some broken cables around here

Varedit ALL the APCs to have them start empty.
StrongDMM_ezxNLnaIQN
I'd give the chair a funny name so people know it's important.

File: code/datums/ruins/beachplanet.dm [hunk 6]
```diff
@@ -28,3 +28,9 @@
 	id = "beach_colony"
 	description = "A abandoned colony. It seems that this colony was abandoned, for a reason or another"
 	suffix = "beach_colony.dmm"
+
+/datum/map_template/ruin/beachplanet/fallenstar
+	name = "Crashed Starwalker"
+	id = "crashed_star"
+	description = "A Crashed Starwalker Class Pirate Ship. It's Crew is Long-Dead."
+	suffix = "beach_crashed_starwalker"
```
Comment (Erikafox) [inline code/datums/ruins/beachplanet.dm]: beach_crashed_starwalker.dmm

**Review by Erikafox — CHANGES_REQUESTED** [review]

**Comment by Erikafox** [comment]
Please do not mark the "I tested this" checkmark if you didn't attempt to spawn it in. That means you didn't test it.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by Erikafox** [comment]
image
This uses the cursed sand.

**Review by Erikafox — APPROVED** [review]

**Pull request merged**